packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -1,12 +1,171 @@
+ * @returns {PendingTxKey}
+ */
+const getPendingTxKey = evidence => {
+  const { amount, forwardingAddress } = evidence.tx;
+  return toPendingTxKey(forwardingAddress, amount);
+};
+
+/**
+ * Get the key for the seenTxs SetStore
+ * @param {CctpTxEvidence} evidence
  */
+const getSeenKey = evidence => {
+  const { txHash, chainId } = evidence;
+  return /** @type {SeenTxKey} */ (JSON.stringify([txHash, chainId]));
+};
 
 /**
+ * The `StatusManager` keeps track of Pending and Seen Transactions
+ * via {@link TxStatus} states, aiding in coordination between the `Advancer`
+ * and `Settler`.
+ *
+ * XXX consider separate facets for `Advancing` and `Settling` capabilities.
+ *
  * @param {Zone} zone
  */
 export const prepareStatusManager = zone => {
-  return zone.exo('Fast USDC Status Manager', undefined, {});
+  /** @type {MapStore<PendingTxKey, PendingTx[]>} */
+  const pendingTxs = zone.mapStore('PendingTxs', {
+    keyShape: M.string(),
+    valueShape: M.arrayOf(PendingTxShape),
+  });
+
+  /** @type {SetStore<SeenTxKey>} */
+  const seenTxs = zone.setStore('SeenTxs', {
+    keyShape: M.string(),
+  });
+
+  /**
+   * Ensures that `txHash+chainId` has not been processed
+   * and adds `txHash+chainId` to `seenTxs` set.
+   *
+   * @param {CctpTxEvidence} evidence
+   */
+  const assertNotSeen = evidence => {
+    const seenKey = getSeenKey(evidence);
+    if (seenTxs.has(seenKey)) {
+      throw makeError(`Transaction already seen: ${q(seenKey)}`);
+    }
+    seenTxs.add(seenKey);
+  };
+
+  return zone.exo(
+    'Fast USDC Status Manager',
+    M.interface('StatusManagerI', {
+      advance: M.call(CctpTxEvidenceShape).returns(M.undefined()),
+      observe: M.call(CctpTxEvidenceShape).returns(M.undefined()),
+      hasPendingSettlement: M.call(M.string(), M.bigint()).returns(M.boolean()),
+      settle: M.call(M.string(), M.bigint()).returns(M.undefined()),
+      view: M.call(M.string(), M.bigint()).returns(M.arrayOf(PendingTxShape)),
+    }),
+    {
+      /**
+       * Add a new transaction with ADVANCED status
+       * @param {CctpTxEvidence} evidence
+       */
+      advance(evidence) {
+        assertNotSeen(evidence);
+        const key = getPendingTxKey(evidence);
+        const entry = { ...evidence, status: TxStatus.Advanced };
+
+        appendToStoredArray(
+          // @ts-expect-error index signature for type 'string' is missing in type 'PendingTx'.
+          pendingTxs,
+          key,
+          entry,
+        );
+      },
+
+      /**
+       * Add a new transaction with OBSERVED status
+       * @param {CctpTxEvidence} evidence
+       */
+      observe(evidence) {
+        assertNotSeen(evidence);
+        const key = getPendingTxKey(evidence);
+        const entry = { ...evidence, status: TxStatus.Observed };
+
+        appendToStoredArray(
+          // @ts-expect-error index signature for type 'string' is missing in type 'PendingTx'.
+          pendingTxs,
+          key,
+          entry,
+        );
```

Comment:
please DRY the duplication with `advance` using a helper in the prepare scope. e.g.
```suggestion
        addTransaction(evidence, TxStatus.Observed);
```